src/main/kotlin/org/rust/cargo/project/configurable/RsProjectConfigurable.kt

```diff
@@ -42,34 +45,39 @@ class RsProjectConfigurable(
             checkBox(RsBundle.message("settings.rust.toolchain.inject.rust.in.doc.comments.checkbox"))
                 .bindSelected(state::doctestInjectionEnabled)
         }
+
+        onApply {
+            settings.modify {
+                it.toolchain = rustProjectSettings.data.toolchain
+                it.explicitPathToStdlib = rustProjectSettings.data.explicitPathToStdlib
+                it.macroExpansionEngine = state.macroExpansionEngine
+                it.doctestInjectionEnabled = state.doctestInjectionEnabled
+            }
+        }
+        onReset {
+            val newData = RustProjectSettingsPanel.Data(
+                toolchain = settings.toolchain ?: RsToolchainBase.suggest(projectDir),
```

Comment:
Note, `RsToolchainBase.suggest(projectDir)` may be quite slow because of WSL
I think you can easily avoid it by using the same approach as in `onIsModified`

Reply:
I'm not sure I get it. We can move `RsToolchainBase.suggest(projectDir)` to a lazy val.